File: Lab-4/sliding_window.cpp
#include <iostream>
#include <string>
using namespace std;

/*
Pseudocode

PROCEDURE sliding_window(sequence, x, y):
    n = len(sequence)
    for i=1 to n by step y do
        if i+x <= n do 
            SHOW sequence[i : i+x]
*/

void sliding_window(string s, int x, int y){
    for (int i = 0; i < s.length(); i+=y){
        if ( (i + x) <= s.length() )
            cout << s.substr(i,x) << endl;
    }
}

int main(){
    
    string s;
    int x;
    int y;

    cin >> s;
    cin >> x;
    cin >> y;


    sliding_window(s, x, y);

    return 0;
}

File: Lab-4/sliding_window.py
from easyinput import read_line,read

'''
Pseudocode

PROCEDURE sliding_window(sequence, x, y):
    n = len(sequence)
    for i=1 to n by step y do
        if i+x <= n do 
            SHOW sequence[i : i+x]
'''

def sliding_window(sequence,x,y):
    for i in range(0, len(sequence),y):
        if i+x <= len(sequence): print(sequence[i:i+x])

s = read_line()
x,y = read(int),read(int)

sliding_window(s, x, y)

#ACGGTAGACCT
'''
ACG
CGG
GGT
GTA
TAG
AGA
GAC
ACC
CCT
'''